src/exercises/day-1/for-loops-bonus.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+fn transpose(matrix: &[&[i32]; 3]) -> [[i32; 3]; 3] {
+    let len_r = matrix[0].len();
```

Comment:
Could you inline this? Since you only use `len_r` and `len_c` once below, we can simply let the compiler optimize this for us (it will know that `matrix[0].len()` doens't change and thus it can compute it only once).

Just for fun, I checked this with the Compiler Explorer: https://rust.godbolt.org/z/vKGTshKGo